hexa/datasets/queue.py

```diff
@@ -52,6 +52,33 @@ def download_file_as_dataframe(
         return pd.read_parquet(download_url)
 
 
+def metadata_profiling(file_content: pd.DataFrame) -> list:
+    for col in file_content.select_dtypes(include=["object"]).columns:
+        file_content[col] = file_content[col].astype("string")
+
+    data_types = file_content.dtypes.apply(str).to_dict()
+    missing_values = file_content.isnull().sum().to_dict()
+    unique_values = file_content.nunique().to_dict()
+    distinct_values = file_content.apply(lambda x: x.nunique(dropna=False)).to_dict()
+    constant_values = (
+        file_content.apply(lambda x: x.nunique() == 1).astype("bool").to_dict()
+    )
+
+    metadata_per_column = [
+        {
+            "column_names": column,
+            "data_types": data_types.get(column, "-"),
```

Comment:
On the backend, having `"-"` as the representation of a None value is not optimal. Better to leave that to None.